upstream dev CI: enable bottleneck again

bottleneck cannot be built with python 3.10. See https://github.com/pydata/xarray/actions/runs/1731371015

```
   gcc -pthread -B /usr/share/miniconda/envs/xarray-tests/compiler_compat -E -I/usr/share/miniconda/envs/xarray-tests/include/python3.1 -I/usr/share/miniconda/envs/xarray-tests/include/python3.1 -o _configtest.i _configtest.c
  _configtest.c:1:10: fatal error: Python.h: No such file or directory
      1 | #include <Python.h>
        |          ^~~~~~~~~~
  compilation terminated.
```
python3.1 huh?

I think https://github.com/pydata/bottleneck/pull/378 might fix but this PR has been sitting around since march.

--- a/ci/install-upstream-wheels.sh
+++ b/ci/install-upstream-wheels.sh
@@ -13,10 +13,10 @@ conda uninstall -y --force \
     rasterio \
     packaging \
     pint \
+    bottleneck \
     sparse \
     h5netcdf \
     xarray
-    # bottleneck \ # re-enable again, see https://github.com/pydata/bottleneck/pull/378
 # to limit the runtime of Upstream CI
 python -m pip install pytest-timeout
 python -m pip install \
@@ -43,8 +43,8 @@ python -m pip install \
     git+https://github.com/mapbox/rasterio \
     git+https://github.com/pypa/packaging \
     git+https://github.com/hgrecco/pint \
+    git+https://github.com/pydata/bottleneck \
     git+https://github.com/pydata/sparse \
     git+https://github.com/intake/filesystem_spec \
     git+https://github.com/SciTools/nc-time-axis \
     git+https://github.com/h5netcdf/h5netcdf
-    # git+https://github.com/pydata/bottleneck \ # re-enable again, see https://github.com/pydata/bottleneck/pull/378

